ouroboros-network/src/Ouroboros/Cardano/PeerSelection/PeerSelectionActions.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE CPP                      #-}
+{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE FlexibleContexts         #-}
+{-# LANGUAGE LambdaCase               #-}
+{-# LANGUAGE NamedFieldPuns           #-}
+{-# LANGUAGE RankNTypes               #-}
+{-# LANGUAGE ScopedTypeVariables      #-}
+{-# LANGUAGE TupleSections            #-}
+
+module Ouroboros.Cardano.PeerSelection.PeerSelectionActions (requestPublicRootPeers) where
+
+
+import Cardano.Network.PeerSelection.Bootstrap (UseBootstrapPeers (..),
+           requiresBootstrapPeers)
+import Cardano.Network.Types (LedgerStateJudgement)
+import Control.Concurrent.Class.MonadSTM.Strict
+import Control.Monad.Class.MonadAsync (MonadAsync)
+import Control.Monad.Class.MonadThrow (Exception, MonadThrow)
+import Control.Monad.Class.MonadTime.SI
+import Control.Tracer (Tracer)
+import Data.Bifunctor (first)
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Data.Set (Set)
+import Data.Set qualified as Set
+import Network.DNS qualified as DNS
+import Ouroboros.Cardano.Network.PublicRootPeers (CardanoExtraPeers (..))
+import Ouroboros.Cardano.Network.Types (CardanoPublicRootPeers)
+import Ouroboros.Network.PeerSelection.LedgerPeers hiding (getLedgerPeers)
+import Ouroboros.Network.PeerSelection.PeerAdvertise (PeerAdvertise (..))
+import Ouroboros.Network.PeerSelection.PeerSelectionActions (getPublicRootPeers)
+import Ouroboros.Network.PeerSelection.PublicRootPeers qualified as PublicRootPeers
+import Ouroboros.Network.PeerSelection.RootPeersDNS (PeerActionsDNS (..))
+import Ouroboros.Network.PeerSelection.RootPeersDNS.DNSSemaphore (DNSSemaphore)
+import Ouroboros.Network.PeerSelection.RootPeersDNS.PublicRootPeers
+
+-- We start by reading the current ledger state judgement, if it is
+-- YoungEnough we only care about fetching for ledger peers, otherwise we
+-- aim to fetch bootstrap peers.
+requestPublicRootPeers
```

Comment:
Where is this function plugged in?

Reply:
In cardano-node

Reply:
So it's similar to [this comment](https://github.com/IntersectMBO/ouroboros-network/pull/5016#discussion_r1935638672).